samples/eShopLite/AppHost/aspire-manifest.json

```diff
@@ -8,6 +8,25 @@
       "parent": "postgres"
     },
     "basketcache": {
+      "type": "container.v0",
+      "image": "redis:latest",
+      "bindings": {
+        "tcp": {
+          "scheme": "tcp",
+          "protocol": "tcp",
+          "transport": "tcp",
+          "containerPort": 6379
+        }
+      },
+      "connectionString": "Host={basketcache.bindings.tcp.host};Port={basketcache.bindings.tcp.port};Username=postgres;Password={basketcache.inputs.password};",
+      "inputs": {
+        "password": {
```

Comment:
Just thinking about how I'm going to do the UX here.  I think right now the best I can do from this is something like:

```
Enter a value for the 'password' property of the 'basketcache' resource:
```

and since it is a secret then read that value as you might read a password (maybe the value isn't echo'd)?

I'm wondering if longer term we want to have a "description" value that would allow a component author to provide more information about what exactly to input here.

I know when we were talking about this sort of stuff a while back (before we had the `inputs` insight) we thought we might want to auto-generate the password for the postgres database ourselves.  Right now, I think the way this is structure we'll have to prompt, since it's not obvious you want the tool to give something for you instead of prompting.

But I do think what you have here is good enough for us to get a first cut of the support.

Reply:
I’d rather have it generated for now

Reply:
> I’d rather have it generated for now, but it needs to be stored right

I'd like us to then change the type requested from "string" to something like "generatedPassword" or something, so that the tool can understand it should generate it instead of prompting (as you might for say an API Key)

> That look ok for storage?

What you outlined here (AesGcm from a passphrase) is basically what I did when I added storing local encrypted stuff in Pulumi.  I suspect it's best you can do without taking a dependency on a managed service like KeyVault (which makes sense for `azd` but perhaps less for you).

Reply:
Ahh cool 😃 
Thanks

Aye Pulumi was what I had in my minds eye when I was writing it. Was going for a similar experience really as I like the secret management there.

That's right. It's mainly a way to keep the local secrets for generated templates safe that can be expanded at runtime. I usually use sealed secrets on my clusters, and 1Password's cli for local files which get secrets at deploy time from directly within 1P vaults, but no guarantee others are using that and didn't want to have to rely on anything third party. 

Think I'll implement AKV at least with this though once first pass is implemented as I have use cases too. All my own (bar a couple of cheap test rigs), and the companies I work withs infrastructure is all Azure based ^^

Reply:
What if we added a field to the input along the lines of:

```json
{
  "inputs": {
    "password": {
      "type": "string",
      "secret": "true",
      "default": {
        "generate": {
          "minLength": 10,
          "maxLength": 20,
          "minAlpha": 1,
          "minSpecial": 1,
          "minNumeric": 1,
        }
      }
    }
  }
}
```

Reply:
Too much for preview2, it needs to be more succinct. Maybe just a minimum length.

Reply:
So this:

```json
{
  "inputs": {
    "password": {
      "type": "string",
      "secret": "true",
      "default": {
        "generate": {
          "minLength": 10
        }
      }
    }
  }
}
```

So we've left room for the others.

Reply:
And we just assume alpha, numeric and special > 1.